github/prci.py

```diff
@@ -247,49 +217,80 @@ def main():
     runner_id = args.ID
     config = args.config
 
-    creds = config['credentials']
-    repo = config['repository']
-    tasks_file = config['tasks_file']
-    whitelist = config['whitelist']
-    no_task_backoff_time = config['no_task_backoff_time']
-
-    logging.config.dictConfig(config['logging'])
-
-    github = github3.login(token=creds['token'])
-    github.session.mount('https://api.github.com',
-                         GitHubAdapter(cache=redis.Redis()))
-
-    repo = github.repository(repo['owner'], repo['name'])
-    task_queue = TaskQueue(repo, tasks_file, JobDispatcher, runner_id,
-                           whitelist)
-
-    handler = ExitHandler()
-    signal.signal(signal.SIGINT, handler.finish)
-    signal.signal(signal.SIGTERM, handler.abort)
-
-    while not handler.done:
-        try:
-            task_queue.create_tasks_for_pulls()
-
-            try:
-                task = next(task_queue)
-            except StopIteration:
-                time.sleep(no_task_backoff_time)
-                continue
-
-            handler.register_task(task)
-            task.execute()
-        except Exception:
-            sentry_report_exception({'module': 'github'})
-            time.sleep(ERROR_BACKOFF_TIME)
-        finally:
-            handler.unregister_task()
-
-            try:
-                check_reboot(repo, creds)
-            except:
-                sentry_report_exception()
-
+    credentials = config["credentials"]
+    repo = config["repository"]
+    tasks_path = config["tasks_file"]
+    whitelist = config["whitelist"]
+    no_task_backoff_time = config["no_task_backoff_time"]
+
+    logging.config.dictConfig(config["logging"])
+
+    exit_handler = ExitHandler()
+    signal.signal(signal.SIGINT, exit_handler.finish)
+    signal.signal(signal.SIGTERM, exit_handler.abort)
+
+    gh = github3.login(token=credentials["token"])
+    session = util.create_session(util.make_headers(credentials["token"]))
+    do_request = partial(util.perform_request, session=session)
+
+    world = World(
+        graphql_request=do_request,
+        github_api=gh,
+        session=session,
+        repo_owner=repo["owner"],
+        repo_name=repo["name"],
+        runner_id=runner_id,
+        tasks_path=tasks_path,
+        whitelist=whitelist
+    )
 
-if __name__ == '__main__':
+    while not exit_handler.done:
+        world.check_graphql_limit()
+
+        response = do_request(
+            query=queries.make_pull_requests_query(
+                world.repo_owner, world.repo_name
+            )
+        )
+
+        if not isinstance(response, dict):
+            logger.error(response)
```

Comment:
`logger.error("Expected github response to be an instance of type 'dict', but it is '%s'. The response: %r", type(response), response))` or split that into two `logger.error()` calls.

Reply:
Removed